app/src/main/java/com/xmartlabs/doapp/ui/TasksListFragment.java

```diff
@@ -0,0 +1,192 @@
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import com.hannesdorfmann.fragmentargs.annotation.FragmentWithArgs;
+import com.xmartlabs.doapp.controller.TaskController;
+import com.xmartlabs.doapp.model.Task;
+import com.xmartlabs.template.R;
+
+import org.threeten.bp.LocalDate;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.OnClick;
+import rx.SingleSubscriber;
+import timber.log.Timber;
+
+/**
+ * Created by santiago on 3/15/17.
+ */
+@FragmentWithArgs
+public class TasksListFragment extends BaseFragment {
+  @BindView(R.id.task_recycler_view)
+  RecyclerView recyclerView;
+  @BindView(R.id.new_task)
+  LinearLayout new_task;
+  @BindView(R.id.title_edit_text)
+  EditText title;
+  @BindView(R.id.description_edit_text)
+  EditText description;
+  @BindView(R.id.fab_button)
+  FloatingActionButton fab_button;
+  @BindView(R.id.header_title)
+  TextView headerTitle;
+
+  @Inject
+  TaskController taskController;
+
+  private TaskAdapter adapter;
+  private List<Task> tasks = Collections.emptyList();
+
+  @LayoutRes
+  @Override
+  protected int getLayoutResId() {
+    return R.layout.list_tasks_fragment;
+  }
+
+  @NonNull
+  @Override
+  public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+    View view = super.onCreateView(inflater, container, savedInstanceState);
+    setpuRecyclerView();
+    //generateTasks();
+    getTasks();
+    new_task.setVisibility(View.GONE);
+    return view;
+  }
+
+  private void setpuRecyclerView() {
+    recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
+    recyclerView.setHasFixedSize(true);
+    adapter = new TaskAdapter(this::onTappedTask);
+    recyclerView.setAdapter(adapter);
+  }
+
+  @OnClick(R.id.fab_button)
+  void onClickedFabButton() {
+    fab_button.setVisibility(View.GONE);
+    new_task.setVisibility(View.VISIBLE);
+  }
+
+  @OnClick(R.id.close)
+  void onClickedCloseImageView() {
+    closeNewTaskView();
+  }
+
+  @OnClick(R.id.add_task)
+  void onClickedAddTextView() {
+    Task task = Task.builder()
+        .title(title.getText().toString().trim())
+        .description(description.getText().toString().trim())
+        .date(LocalDate.now())
+        .isFinished(false)
+        .build();
+    insertTask(task);
+    closeNewTaskView();
+  }
+
+  private void generateTasks() {
+    List<Task> taskList = new ArrayList<>();
+    for (int i = 0; i < 5; i++) {
+      boolean isFinished = i % 7 == 0;
+      String description = i % 21 == 0 ? "" : "Descrption task #" + i;
+      Task task = Task.builder()
+          .title("Task #" + i)
+          .description(description)
+          .date(LocalDate.now())
+          .isFinished(isFinished)
+          .build();
+      taskList.add(task);
+    }
+    tasks = taskList;
+  }
+
+  private void setFieldsEmpty() {
```

Comment:
which fields are these? The add new task fields? If so, make this method name reflect that